### ISSUE TITLE
Running into an error while trying to build OP-TEE for the HiKey board

### ISSUE THREAD

**Issue by jsherman3** [OP-TEE/optee_os#846]
Hello, I'm currently just trying to build OP-TEE for the HiKey board using the README instructions.

I have downloaded all of the prerequisite software required.
I entered the following commands:
mkdir -p $HOME/devel/optee
cd $HOME/devel/optee
repo init -u https://github.com/OP-TEE/manifest.git -m hikey.xml
repo sync
cd build
make toolchains
make all

When I run the make all, it starts compiling, but I run into the following error:

g++ -c -DPCCTS_USE_NAMESPACE_STD -I Pccts/h -I .. -I ../Include/Common -I ../Include/ -I ../Include/IndustryStandard -I ../Common/ -I .. -I . -I ../Include/X64/   Pccts/h/AParser.cpp -o AParser.o
make[3]: g++: Command not found
make[3]: *** [AParser.o] Error 127
make[3]: Leaving directory `/home/jsherman/devel/optee/edk2/BaseTools/Source/C/VfrCompile'
make[2]: *** [VfrCompile] Error 2
make[2]: Leaving directory `/home/jsherman/devel/optee/edk2/BaseTools/Source/C'
make[1]: *** [Source/C] Error 2
make[1]: Leaving directory `/home/jsherman/devel/optee/edk2/BaseTools'
make: *** [/Conf/target.txt] Error 2

I noticed it says g++ command not found, but all the toolchains are present in the ~/devel/optee/toolchains/aarch64/bin folder

**Comment by jbech-linaro** [comment]
Hi @jsherman3 ,

**TL;DR;** I cannot reproduce the error.

**Long version**
Have you done any update recently? Not that long ago we changed to the default toolchains in use there there were various changes in various gits.

Having that said, I've just build from scratch (repo init, sync, etc ...), i.e., latest available this evening. For your reference I provide the list here.
```bash
aes-perf -- 8d17a26 TA: remove single instance flag
arm-trusted-firmware -- c006778 Merge pull request #2 from vishalbhoj/hikey
build -- 69a8a37 gcc5: Update to use gcc5
burn-boot -- db522b1 Merge pull request #5 from doanac/regression
busybox -- dbf5a6d blkdiscard: new applet
edk2 -- 7da6cdf Merge pull request #70 from vishalbhoj/hikey
gen_rootfs -- 44a44ba Rely on CROSS_COMPILE instead of CC_DIR
grub -- b524fa2 net: translate pxe prefix to tftp when checking for self-load
l-loader -- 6b784ad Merge pull request #10 from fboudra/master
linux -- 2faeddf optee: avoid delayed release of shm
optee_client -- 17d1add Check flags before allocating shared memory.
optee_os -- 3f19eae ARM Juno GCC5 support note.
optee_test -- 5b41fbf Change parameters to TEEC_AllocateSharedMemory
patches_hikey -- 5f4024c Change libteec path according to gcc5
sha-perf -- 57516de TA: remove single instance flag
strace -- 87baaec tests/attach-p-cmd-p.c: cleanup
```

And the particular line that fails in your setup compiles without any issues on my setup.
```c
...
make[3]: Entering directory `/home/joakim.bech/devel/optee_projects/hikey/edk2/BaseTools/Source/C/VfrCompile'
g++ -c -DPCCTS_USE_NAMESPACE_STD -I Pccts/h -I .. -I ../Include/Common -I ../Include/ -I ../Include/IndustryStandard -I ../Common/ -I .. -I . -I ../Include/X64/   Pccts/h/AParser.cpp -o AParser.o
g++ -c -DPCCTS_USE_NAMESPACE_STD -I Pccts/h -I .. -I ../Include/Common -I ../Include/ -I ../Include/IndustryStandard -I ../Common/ -I .. -I . -I ../Include/X64/   Pccts/h/DLexerBase.cpp -o DLexerBase.o
g++ -c -DPCCTS_USE_NAMESPACE_STD -I Pccts/h -I .. -I ../Include/Common -I ../Include/ -I ../Include/IndustryStandard -I ../Common/ -I .. -I . -I ../Include/X64/   Pccts/h/ATokenBuffer.cpp -o ATokenBuffer.o
...
```

I've uploaded the full log [here](https://transfer.sh/PSDlB/build.log) (valid 14-days from today).

**Comment by jforissier** [comment]
@jsherman3 'g++: command not found' is pretty explicit, no? Install the C++ compiler -- the x86/x86_64 one, not the *cross* compiler (your ARM toolchain is probably fine).
That should be `sudo apt-get install g++`, I think.
